cmd/infra/powervs/create.go

```diff
@@ -388,28 +384,44 @@ func (infra *Infra) SetupInfra(ctx context.Context, options *CreateInfraOptions)
 		return fmt.Errorf("cloud connection is not up: %w", err)
 	}
 
+	// setupSecrets need parameter cloudInstanceId, hence invoked after setupPowerVSCloudInstance
+	if err = infra.setupSecrets(options); err != nil {
+		return fmt.Errorf("error setup secrets: %w", err)
+	}
+
 	log(options.InfraID).Info("Setup infra completed in", "duration", time.Since(startTime).String())
 	return nil
 }
 
 // setupSecrets generate secrets for control plane components
 func (infra *Infra) setupSecrets(options *CreateInfraOptions) error {
 	var err error
+	var powerVsCloudInstanceID string
+
+	if options.CloudInstanceID != "" {
+		powerVsCloudInstanceID = options.CloudInstanceID
+	} else if infra.CloudInstanceID != "" {
+		powerVsCloudInstanceID = infra.CloudInstanceID
+	} else {
+		return fmt.Errorf("error setup secrets: unable to limit access scope to instance level: cloud instance not found")
```

Comment:
Already you are appending `error setup secrets` in error message in caller.
```suggestion
		return fmt.Errorf("unable to limit access scope to instance level: cloud instance not found")
```